### DISCUSSION TITLE
Assigning a Role to user in runtime db Connection.

### DISCUSSION THREAD

**Discussion by sanishkamboj** [spatie/laravel-permission#2006] [Q&A]
I want to assign a role to a user created in a runtime connected DB instance. I want to override the model_type during assigning a role to user. Please let me know How I can do that ?

**Comment by PaolaRuby** [comment]
Override the method with your own implementation on assigning roles

**Reply by sanishkamboj** [reply]
I tried to add entry directly in model_has_roles table via custom query. But it gives me error of foreign key. I try to modify the function in vendors as well. but that is not a good practice. Please let me know What else I can do to solve this issues. 
I am using
`$user->assignRole($role);`

**Reply by PaolaRuby** [reply]
Never modify the methods from vendor, only override them, also if you want to change `model_type` you have to override roles relation morphToMany to hasManyThrough but that makes `model_type` useless(I don't undertand why you wanna do something like that)

**Comment by erikn69** [comment]
Maybe this is was you are looking for: https://github.com/spatie/laravel-permission/discussions/2012#discussioncomment-2134652

use `Relation::morphMap` to alias `model_type` to a standard  name
[enforcing-morph-maps-in-laravel](https://laravel-news.com/enforcing-morph-maps-in-laravel)
[custom-polymorphic-types](https://laravel.com/docs/master/eloquent-relationships#custom-polymorphic-types)
